plugin-server/src/ingestion/ingestion-consumer.ts

```diff
@@ -138,16 +140,25 @@ export class IngestionConsumer {
         this.ingestionWarningLimiter = new MemoryRateLimiter(1, 1.0 / 3600)
         this.hogTransformer = new HogTransformerService(hub)
 
-        this.personStore = new MeasuringPersonsStore(this.hub.db, {
-            personCacheEnabledForUpdates: this.hub.PERSON_CACHE_ENABLED_FOR_UPDATES,
-            personCacheEnabledForChecks: this.hub.PERSON_CACHE_ENABLED_FOR_CHECKS,
-        })
+        if (this.hub.PERSON_BATCH_WRITING_ENABLED) {
+            this.personStore = new BatchWritingPersonsStore(this.hub.db, {
+                optimisticUpdatesEnabled: this.hub.PERSON_BATCH_WRITING_OPTIMISTIC_UPDATES_ENABLED,
+                maxConcurrentUpdates: this.hub.PERSON_BATCH_WRITING_MAX_CONCURRENT_UPDATES,
+                maxOptimisticUpdateRetries: this.hub.PERSON_BATCH_WRITING_MAX_OPTIMISTIC_UPDATE_RETRIES,
+                optimisticUpdateRetryInterval: this.hub.PERSON_BATCH_WRITING_OPTIMISTIC_UPDATE_RETRY_INTERVAL_MS,
+            })
+        } else {
+            this.personStore = new MeasuringPersonsStore(this.hub.db, {
+                personCacheEnabledForUpdates: this.hub.PERSON_CACHE_ENABLED_FOR_UPDATES,
+                personCacheEnabledForChecks: this.hub.PERSON_CACHE_ENABLED_FOR_CHECKS,
+            })
+        }
```

Comment:
suggestion: I'm thinking how to roll this out safely – because this is a pretty significant change in a very sensitive place, I'd feel better if we could get some comparison metrics before switching on the writes to the database.

A pattern that worked very well for me in the past was having a "shadowing" setup, where we'd call all operations on both implementations, but only one of them (initially the `MeasuringPersonStore`) would be writing to the database. We could compare what fetch queries return in both implementations and maybe also what inputs go into the updates.

It would also be good to have a percentage rollout for this. 

WDYT?